.github/workflows/lint-golang.yml

```diff
@@ -42,12 +42,12 @@ jobs:
     - name: go mod tidy (cluster-to-redpanda-migration)
       working-directory: src/go/cluster-to-redpanda-migration
       run: |
-        go mod tidy
+        GOPRIVATE=github.com/redpanda-data/flux-controller-shim go mod tidy
         git diff --exit-code -- go.mod go.sum
```

Comment:
i think another step is needed to setup git to access the private repos:
```yaml
    - name: configure git
      env:
        GH_ACCESS_TOKEN: ${{ secrets.VBOT_GITHUB_API_TOKEN }}
      run: git config --global "url.https://$GH_ACCESS_TOKEN@github.com/.insteadOf" https://github.com/
    - name: go mod tidy (cluster-to-redpanda-migration)
      working-directory: src/go/cluster-to-redpanda-migration
      env:
        GOPRIVATE: github.com/redpanda-data/flux-controller-shim
      run: |
        go mod tidy
        git diff --exit-code -- go.mod go.sum
```
based on https://blog.fabianmendez.dev/how-to-use-private-go-module-in-github-actions

Reply:
I think you are right @andrewhsu , can you make that a code recommendation that can be added in a commit?

Reply:
Ok I will try it out